packages/web-components/src/tree-item/tree-item.template.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import { children, elements, html, when } from '@microsoft/fast-element';
+import { FluentDesignSystem } from '../fluent-design-system.js';
+import type { TreeItem } from './tree-item.js';
+
+// We don't put the icon into the icons/index.ts file because
+// this icon is the default chevron icon of the tree-item component,
+// it would be contributed to the upstream later.
+const chevronIcon = html`
+  <svg width="12" height="12" xmlns="http://www.w3.org/2000/svg" fill="currentColor">
+    <path
+      d="M4.65 2.15a.5.5 0 000 .7L7.79 6 4.65 9.15a.5.5 0 10.7.7l3.5-3.5a.5.5 0 000-.7l-3.5-3.5a.5.5 0 00-.7 0z"
+    ></path>
+  </svg>
+`;
+
+export const template = html<TreeItem>`
+  <template
+    role="treeitem"
+    tabindex="-1"
+    slot="${x => (x.isNestedItem ? 'item' : void 0)}"
+    class="${x => x.calculatedClassName}"
+    aria-disabled="${x => x.disabled}"
+    aria-selected="${
+      // The value of the x could be null due to a bug in the FAST itself...
+      x => !!x?.selected
+    }"
+    aria-expanded="${x => (x?.childTreeItems?.length ? x.expanded : void 0)}"
```

Comment:
We've made quite a few updates to leverage more of the modern web's capabilities - can this be updated to use `ElementInternals` and align with the rest of the package?

Reply:
fixed